numpy/core/src/common/npy_cpu_features.h

```diff
@@ -0,0 +1,116 @@
+#ifndef _NPY_CPU_FEATURES_H_
+#define _NPY_CPU_FEATURES_H_
+
+#include "numpy/npy_common.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum npy_cpu_features
+{
+    NPY_CPU_FEATURE_NONE = 0,
+    // X86
+    NPY_CPU_FEATURE_MMX               = 1,
+    NPY_CPU_FEATURE_SSE               = 2,
+    NPY_CPU_FEATURE_SSE2              = 3,
+    NPY_CPU_FEATURE_SSE3              = 4,
+    NPY_CPU_FEATURE_SSSE3             = 5,
+    NPY_CPU_FEATURE_SSE41             = 6,
+    NPY_CPU_FEATURE_POPCNT            = 7,
+    NPY_CPU_FEATURE_SSE42             = 8,
+    NPY_CPU_FEATURE_AVX               = 9,
+    NPY_CPU_FEATURE_F16C              = 10,
+    NPY_CPU_FEATURE_FMA3              = 11,
+    NPY_CPU_FEATURE_XOP               = 12,
+    NPY_CPU_FEATURE_FMA4              = 13,
+    NPY_CPU_FEATURE_AVX2              = 14,
+    NPY_CPU_FEATURE_FMA               = 14, // alias to provide backward compatibility
```

Comment:
Where do we have any backward compatibility problems?

Reply:
Here the situation,
AMD chips "Piledriver" has support for `FMA3` but without `AVX2` support.
All intel and AMD chips that support `AVX2`  also support `FMA3`.

And the current SIMD code requires `AVX2` and `FMA3` with a feature name `FMA`, so I had to add `FMA` for combability, and it seems I was wrong since mapping `FMA` to `AVX2` will not be safe because according to "Peter Cordes" in his comment at [stackoverflow.](https://stackoverflow.com/a/55763975), "there is a real x86 CPU that supports AVX2 but without FMA3"  manufactured by "VIA".

So now `FMA` is added for compatibility reasons and it detects both `AVX2` and `FMA3`.